src/Spec-Tools/ChangeSorterPresenter.class.st

```diff
@@ -347,6 +354,12 @@ ChangeSorterPresenter >> forgetMessage [
 		self setSelectedClassIndex: self selectedClassIndex ]
 ]
 
+{ #category : #api }
+ChangeSorterPresenter >> initialExtent [
```

Comment:
We are deprecating this way to set the initial extent of the window. 

The way we will try to enforce is to add a method `initializeWindow:` taking a window presenter as a parameter and to set it to this presenter. The goal is to remove window state from all composable presenters.

Reply:
Well how can I know it?
This is not explained anywhere! 
What we should do then is to change all the initialExtent to use initialWindow: 
now I could not understand why the previous initialWindow: method was not doing a super call.

Then I do not understand why self title: 'HJKJL' does not update the window title. 
So this is confusing.

Reply:
How should I write initializeWindow: 
because I do not get it. When I read the following code I see that the initialExtent is invoked.

So again Spec20 is not correctly communicating with me. 





initializeWindow: aWindowPresenter
	"override this to set window values before opening. 
	 You may want to add a menu, a toolbar or a statusbar"
	
	aWindowPresenter 
		title: self title;
		initialExtent: self initialExtent;
		windowIcon: self windowIcon;
		askOkToClose: self askOkToClose;
		aboutText: self aboutText

Reply:
then if I should write

initializeWindow: aWindowPresenter

    super initializeWindow: aWindowPresenter.
    aWindowPresenter initialExtent: 500@ 350.

instead of 

initialExtent
    ^ 500@ 350 

I do not really understand the gain because the framework puts the burden on 
its user. So to me this is pretty bad design. 
For example I can forget to call super.

Reply:
Cyril I will integrate it now and we will see.